Allow healthcheck failures

sroze opened the pull request:
This PR introduce an `allowed_failures` parameter for applications' healthcheck. This is the number of allowed failures before that the health check consider the application as unhealthy (the default value is for sure 0)

This is especially interesting when sometimes there's some need of asynchronous process to be started before the application can have its normal behavior.

andrewsmedina commented on provision/docker/healthcheck.go:
```diff
@@ -71,6 +71,14 @@ func runHealthcheck(cont *container, w io.Writer) error {
 			return err
 		}
 	}
+	var allowed_failures int
```
You should use camel case instead of underscores.

andrewsmedina commented on provision/docker/healthcheck.go:
```diff
@@ -71,6 +71,14 @@ func runHealthcheck(cont *container, w io.Writer) error {
+		allowed_failures = val
+	case float64:
+		allowed_failures = int(val)
+	default:
```
you can remove this line

sroze replied:
So I also removed this default for `status`.

andrewsmedina commented:
@sroze can you update the docs about `healthcheck` (http://docs.tsuru.io/en/master/using/tsuru.yaml.html#healthcheck) ?

sroze commented:
@andrewsmedina I updated previous commit base on your comments and added the documentation bullet.

sroze commented:
@andrewsmedina I updated previous commit base on your comments and added the documentation bullet (and rebuilt this morning as previous Travis build failed to launch)

andrewsmedina commented:
@sroze thank you!
